experiment/Commonfunctions/commonfunctions-exercise.js

```diff
@@ -70,55 +57,150 @@ function resetTraversal() {
         document.getElementById("bft").style.color = '#FFFFFF';
         document.getElementById("bft").style.borderColor = "#3297CF";
     }
-  }
-
+}
+window.resetTraversal=resetTraversal;
 
-  function submit(){
-        if(index==(tree_traversal.sequence_list.length-1)){
-        if(tree_traversal.error==0){
-        document.getElementById("commentbox").style.display = "block";
-        document.getElementById("comments").innerHTML = 'Traversal Complete.Your Traversal is correct!'
-        }
-        else{
-        document.getElementById("commentbox").style.display = "block";
-        document.getElementById("comments").innerHTML = 'Traversal Complete.Your Traversal is incorrect.Try again! \n Re-attempt practice section.'
-        }
-        }
-        else{
-        document.getElementById("commentbox").style.display = "block";
-        document.getElementById("comments").innerHTML = 'Traversal Incomplete.Your Traversal is incorrect.Try again! \n Re-attempt practice section.'
-        }
-        }
 
-function check(d){
-if(tree_traversal.traversal_selected==false){
-      alert("Select a traversal first.");
-      return;
-    }
-    index+=1; 
-    if(index==(tree_traversal.sequence_list.length-1)){
-    visitElement(d,0)
-    if(tree_traversal.error==0){
-    document.getElementById("commentbox").style.display = "block";
-    document.getElementById("comments").innerHTML = 'Traversal Complete.Your Traversal is correct!'
-    }
-    else{
-    document.getElementById("commentbox").style.display = "block";
-    document.getElementById("comments").innerHTML = 'Traversal Complete.Your Traversal is incorrect.Try again! \n Re-attempt practice section.'
+export function generateGraph(){
+    
+    let textTraversal=document.getElementById('tree');
+    d3.select("svg").remove();
+    changesvg(d3.select("#tree").append("svg")
+    .attr("id","grap") 
+    .attr("width", tree_traversal.width + tree_traversal.margin.right + tree_traversal.margin.left)
+    .attr("height", tree_traversal.height + tree_traversal.margin.top + tree_traversal.margin.bottom)
+    .attr('viewBox','-70 -15 700 530')
+    .append("g")
+    .attr("transform", "translate(" + tree_traversal.margin.left + "," + tree_traversal.margin.top + ")"));
+    var random=Math.floor(Math.random() * 100);  
+    random=(random%treeData.length);
+    changeroot(treeData[random]);
+    update(treeData[random]);
+    if(document.getElementById("preorder")){
+  
+        resetTraversal();
+        tree_traversal.sequence_list = [];
+        preorder();
     }
+    if(document.getElementById("bft"))bft();
+}
+
+export function submit(){
+    // Comparing and checking the equality of the two lists 
+    if( Array.from(node_map.values()).length!==tree_traversal.sequence_list.length){
+
+
+       tree_traversal.error=1;
```

Comment:
Unfortunately since the practice and demo operate differently compared to the exercise I can't implement this change. I understand that boolean would be better. If I however, make the error variable a boolean, other code would be affected. Again, demo, practice and exercise import from the same global variable file.